### PR TITLE
Filter out DataBinding expressions when extracting resources from xml files

### PR DESCRIPTION
**Pull request by nsk-mironov**
I've tried running `buildHealth` with `0.79.0` version of the plugin and got the same exception as in #531 and #542. The only difference that it has nothing to do with `@null` in my case and is actually caused by [data binding expressions](https://developer.android.com/topic/libraries/data-binding/expressions) like `binding:subtitle="@{model.subtitle}"`.

The fix from #534 actually prevents the exception in my case, but it doesn't cover all databinding related scenarios. For example, expressions like `binding:outlineRadius="@{@dimen/radius_normal}"` are still parsed as a valid `AttrRef(type = "{@dimen", id = "radius_normal}")` object.

This PR introduces the following changes:
- `AttrRef.from` now always returns `null` for all data binding expressions.
- `ResSpec.groovy` runs one more case which is specially dedicated to data binding expressions.

### CODE REVIEW

File: src/functionalTest/kotlin/com/autonomousapps/fixtures/DataBindingWithExpressionsProject.kt [hunk 1]
```diff
@@ -0,0 +1,48 @@
+package com.autonomousapps.fixtures
+
+import com.autonomousapps.advice.Advice
+
+class DataBindingWithExpressionsProject(
```
Comment (autonomousapps) [inline src/functionalTest/kotlin/com/autonomousapps/fixtures/DataBindingWithExpressionsProject.kt:5]: Please rewrite this fixture to use a similar style to `AttrResWithNullProject` from the spec just above your new one. This is the first style of fixture used in this project and... I hate it. I'd like to eliminate it entirely at some point.

File: src/main/kotlin/com/autonomousapps/model/Source.kt [hunk 2]
```diff
@@ -71,14 +71,16 @@ data class AndroidResSource(
...
+       * 3. They're not a data binding expression (don't start with `@{` and end with `}`)
+       * 4. Their value starts with `?`, like `?themeColor`.
+       * 5. Their value starts with `@`, like `@drawable/`.
        *
        * Will return `null` if the map entry doesn't match an expected pattern.
        */
       fun from(mapEntry: Map.Entry<String, String>): AttrRef? {
         if (mapEntry.isId()) return null
         if (mapEntry.isToolsAttr()) return null
+        if (mapEntry.isDataBindingExpression()) return null
```
Comment (autonomousapps) [inline src/main/kotlin/com/autonomousapps/model/Source.kt:83]: Thank you for updating this in both locations!

**Review by autonomousapps — CHANGES_REQUESTED** [review]
Thank you for this contribution! Very much appreciated. Only one change I'd like to see.

File: src/functionalTest/kotlin/com/autonomousapps/fixtures/DataBindingWithExpressionsProject.kt (hunk 1 above)
Comment (nsk-mironov) [reply src/functionalTest/kotlin/com/autonomousapps/fixtures/DataBindingWithExpressionsProject.kt]: Sure, not a problem. Please check my latest commit with an updated `DataBindingWithExpressionsProject` implementation.

**Review by autonomousapps — APPROVED** [review]
LGTM!

**Pull request merged**